storage/src/merkle_storage.rs

```diff
@@ -1059,201 +847,48 @@ impl MerkleStorage {
     /// * `root_hash` - hash of Tree to modify
     /// * `key` - path under which the changes takes place
     /// * `new_node` - None for deletion, Some for inserting a hash under the key.
+    /// Get a new tree with `new_entry_hash` put under given `key`.
+    ///
+    /// # Arguments
+    ///
+    /// * `root` - Tree to modify
+    /// * `key` - path under which the changes takes place
+    /// * `new_entry_hash` - None for deletion, Some for inserting a hash under the key.
     fn compute_new_root_with_change(
         &mut self,
-        root_hash: &EntryHash,
+        root: &Tree,
         key: &[String],
         new_node: Option<Node>,
     ) -> Result<EntryHash, MerkleError> {
-        assert_eq!(key.is_empty(), false);
         if key.is_empty() {
```

Comment:
If I am understanding this right, if the key is empty, we just replace the tree with a new one, right? would be good to have a comment about that here.